lib/stateTransition/AbstractStateTransition.js

```diff
@@ -49,6 +77,154 @@ class AbstractStateTransition {
   hash() {
     return hash(this.serialize()).toString('hex');
   }
+
+  /**
+   *  Returns signature
+   *
+   * @return {string|null}
+   */
+  getSignature() {
+    return this.signature;
+  }
+
+  /**
+   * Returns public key id
+   *
+   * @returns {number|null}
+   */
+  getSignaturePublicKeyId() {
+    return this.signaturePublicKeyId;
+  }
+
+  /**
+   * Sign data with privateKey
+   *
+   * @param {IdentityPublicKey} identityPublicKey
+   * @param {string} privateKey
+   * @return {AbstractStateTransition}
+   */
+  sign(identityPublicKey, privateKey) {
+    const data = this.serialize({ skipSignature: true });
+    const privateKeyInHex = Buffer
```

Comment:
Do we always suppose that private key is encoded with base64?

Reply:
@shuplenkov updated ^

Reply:
PrivateKey accepts only base58 string. We can get confused with that...

Reply:
Actually, it accepts also buffer, private key model, Uint8Array. Base58 is a common encoding for Dash, we use it for identities and everywhere in dash core.

Reply:
Okay, so I've checked out how dashcore-lib works internally, and it appears that when you create an instance of `PrivateKey` from `Buffer`, then when you call `.toPublicKey();` It will create an uncompressed public key, and when you do this with a hex, it will create a compressed one. It doesn't make any sense to me so far. Compressed pubkey key has only X point in it, while uncompressed has X, Y, and a network prefix.

When we are speaking about the public key Y coordinate, it generally not necessary needed, as it's possible to calculate Y coordinate from the X coordinate. When performing validation, if the key is compressed, it means slower validation, since you need to compute Y before performing an actual validation.

Here's a Bitcoin developer Pieter Wuille gives an explanation about this if you're interested: https://bitcoin.stackexchange.com/questions/19666/why-do-keys-need-both-x-and-y-coordinates-if-x-can-be-solved-for-y-using-the-cu#19669

Interestingly enough, in DID, if the key stored in the hex format, it's always compressed, and uncompressed otherwise: check out the draft on ECDSA keys, especially the last line in the example paragraph: https://w3c-dvcg.github.io/lds-ecdsa-secp256k1-2019/#examples